### PR TITLE
(PC-6169) intégrer le header bénéficiaire dans la page profile

### PR DESCRIPTION
**Pull request by AnoukHello**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6169

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

Screenshot from 2021-02-12 09-43-30

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/api/gen/.swagger-codegen/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-3.0.24
\ No newline at end of file
+3.0.23
```
Comment (blixit) [inline src/api/gen/.swagger-codegen/VERSION:1]: problème de version retrograde. solution dans cette conversation: https://passcultureteam.slack.com/archives/G01C85B6DHA/p1612282625008300

File: src/features/profile/components/ProfileHeader.tsx [hunk 2]
```diff
@@ -20,18 +16,24 @@ export function ProfileHeader(props: ProfileHeaderProps) {
...
 
   if (!user) {
-    // afficher le header non connecté
+    // TODO(PC-6372) afficher le header non connecté
     return null
   }
 
-  if (props.user?.isBeneficiary) {
+  if (user.isBeneficiary) {
+    console.warn(user)
```
Comment (blixit) [inline src/features/profile/components/ProfileHeader.tsx:24]: console.warn

File: src/features/profile/components/ProfileHeader.tsx [hunk 3]
```diff
@@ -20,18 +16,24 @@ export function ProfileHeader(props: ProfileHeaderProps) {
...
+    const depositVersion = user.depositVersion && user.depositVersion === 1 ? 1 : 2
```
Comment (blixit) [inline src/features/profile/components/ProfileHeader.tsx:25]: si le user est bénéficiaire depositVersion est toujours défini. on devrait le refléter dans le typage pour plus de robustesse

File: src/features/profile/components/BeneficiaryHeader.tsx [hunk 4]
```diff
@@ -46,9 +45,11 @@ export function BeneficiaryHeader(props: PropsWithChildren<BeneficiaryHeaderProp
         {/* eslint-disable-next-line react-native/no-raw-text */}
         <Typo.Hero color={ColorsEnum.WHITE}>{`${props.remainingCredit} €`}</Typo.Hero>
         <Spacer.Column numberOfSpaces={2} />
-        <Typo.Caption color={ColorsEnum.WHITE}>
-          {_(t`crédit valable jusqu'au ${creditExpiracyDate}`)}
-        </Typo.Caption>
+        {props.depositExpirationDate && (
```
[the remark this answers is not in the document]
Comment (AnoukHello) [reply src/features/profile/components/BeneficiaryHeader.tsx:48]: Le type de cette propriété est défini par le backend (UserProfileResponse)

File: src/features/profile/components/ProfileHeader.tsx (hunk 3 above)
Comment (AnoukHello) [reply src/features/profile/components/ProfileHeader.tsx:25]: Le type de cette propriété est défini par le backend (UserProfileResponse)

**Pull request merged**